src/main/java/school/lemon/changerequest/java/introduction/hw2/Task10.java

```diff
@@ -1,7 +1,20 @@
 package school.lemon.changerequest.java.introduction.hw2;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class Task10 {
     public static String trim(String text) {
-        return "";
+
+        if (text == null)
+            return null;
+        String res = "";
+
+        Pattern p = Pattern.compile("(\\S.*\\S)|(\\S)");
```

Comment:
It's possible to do with only one regex group, e.g. you can use following regex:
`\s*(\S.+\S)\s*`

Reply:
This regex doesn`t work when the string contain only one character

Reply:
@Mar-Alex you're right.

I've adjusted it a bit: `\s*(\S?.*\S)\s*`